chain/chain/src/resharding.rs

```diff
@@ -181,15 +201,49 @@ impl Chain {
         sync_hash: &CryptoHash,
         state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
     ) -> Result<(), Error> {
-        let prev_hash = *self.get_block_header(sync_hash)?.prev_hash();
+        let state_roots = state_roots?;
+        let block_header = self.get_block_header(sync_hash)?;
+        let prev_hash = block_header.prev_hash();
+
+        let child_shard_uids: Vec<_> = state_roots.keys().collect();
+        self.initialize_flat_storage(&prev_hash, &child_shard_uids)?;
```

Comment:
When this method is called the flat storage should already be populated with all of the state data for the new shards. Is it? If so can you remind me where this happens?

Reply:
This happens as a part of `tries.add_values_to_split_states` function call in PR https://github.com/near/nearcore/pull/9421

Reply:
Nice, thanks, I missed that.